### PR TITLE
LPD-1735 The external reference code for schema should be unique

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPD-1735

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:test:sf

**Comment by magjed4289** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISchemaRelevantObjectEntryModelListener.java [hunk 1]
```diff
@@ -107,6 +108,40 @@ private void _validate(ObjectEntry objectEntry) {
...
+							objectEntry.getExternalReferenceCode(),
+							"' and r_apiApplicationToAPISchemas_c_",
+							"apiApplicationId ne '",
+							originalObjectEntryValues.get(
+								"r_apiApplicationToAPISchemas_c_" +
+									"apiApplicationId"),
+							"'"),
+						"L_API_SCHEMA"))) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
```
Comment (albertojml) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISchemaRelevantObjectEntryModelListener.java:132]: I dont know if we should return "The external reference code is already in use", which is the message returned by the objects layer. Just to mantain consistency. It's just a question for debating, I'm not sure about it...

**Review by albertojml — COMMENTED** [review]
Just a small question.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
@magjed4289 this really applies to every object related to another object. The fix must be applied everywhere as discussed on slack.

**Comment by 4lejandrito** [comment]
Closing as won't fix. See the jira ticket for more info.

**Pull request closed without merging**